service/src/main/java/org/mskcc/cmo/metadb/service/impl/SampleServiceImpl.java

```diff
@@ -107,12 +111,14 @@ public MetadbSample getMetadbSample(UUID metadbSampleId) throws Exception {
     public MetadbSample getMetadbSampleByRequestAndAlias(String requestId, SampleAlias igoId)
             throws Exception {
         MetadbSample sample = sampleRepository.findSampleByRequestAndIgoId(requestId,
-                igoId.getSampleId());
+                igoId.getValue());
+        if (sample == null) {
+            return null;
+        }
         sample.setSampleMetadataList(sampleRepository
                 .findSampleMetadataListBySampleId(sample.getMetaDbSampleId()));
         for (SampleMetadata s : sample.getSampleMetadataList()) {
             s.setMetaDbPatientId(patientService.getPatientIdBySample(sample.getMetaDbSampleId()));
-            s.setMetaDbSampleId(sample.getMetaDbSampleId());
```

Comment:
maybe instead of metadb patient id being set just above this we should just be fetching the actual `MetadbPatient` and calling `sample.setPatient(...)`

I feel it's a little redundant to keep access to metadb patient id on the sample object level while we also have access to the `MetadbPatient` itself at the sample object level as well

Reply:
still need to fetch and set the sample's patient here 

[reference](https://github.com/mskcc/cmo-metadb/pull/468/files#diff-14fbd1dc6e948619c6d34efb8f41005965101954a8895440857979d916eb6b57L130)